tests/test_syntax/test_sequential2.py

```diff
@@ -63,3 +65,21 @@ def __call__(self, I: m.Bits[4]) -> m.Bits[4]:
     m.compile("build/TestSequential2Hierarchy", Bar)
     assert check_files_equal(__file__, f"build/TestSequential2Hierarchy.v",
                              f"gold/TestSequential2Hierarchy.v")
+
+
+def test_sequential2_pre_unroll():
```

Comment:
I didn't have an explicit usecase in mind for these hooks (I just figured I need them at some point) but this is a great example!

Reply:
This revealed some nuances with supporting the setattr/getattr interfaces and the general python code in __init__ (attribute of type list in this case).  I added basic support for the test, but I'm sure there's more general cases where this fails, but I figure we can address them in time as we see the examples.  But beware of this possible issue and let me know if you run into it (sequential attributes more complicated that a register or list of registers or nested list)